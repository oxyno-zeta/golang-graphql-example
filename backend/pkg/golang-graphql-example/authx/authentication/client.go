package authentication

import (
	"net/url"
	"regexp"

	"github.com/gin-gonic/gin"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/config"
)

//go:generate mockgen -destination=./mocks/mock_Client.go -package=mocks github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/authx/authentication Client
type Client interface {
	// Middleware will redirect authentication to basic auth or OIDC depending on request path and resources declared.
	Middleware(unauthorizedPathRegexList []*regexp.Regexp) gin.HandlerFunc
	// OIDCEndpoints will set OpenID Connect endpoints for authentication and callback.
	OIDCEndpoints(router gin.IRouter) error
}

type providerEndpointsClaims struct {
	EndSessionEndpointURL *url.URL
	EndSessionEndpoint    string `json:"end_session_endpoint"`
}

func NewService(cfgManager config.Manager) Client {
	return &service{
		cfgManager: cfgManager,
	}
}
