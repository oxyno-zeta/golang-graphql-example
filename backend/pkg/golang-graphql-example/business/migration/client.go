package migration

import "github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/database"

// Client will represent the service that will migrate database.
type Client interface {
	// Migrate.
	Migrate() error
}

func New(dbSvc database.DB) Client {
	return &service{dbSvc: dbSvc}
}
