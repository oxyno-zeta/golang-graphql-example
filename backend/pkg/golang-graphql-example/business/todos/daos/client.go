package daos

import (
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/business/todos/models"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/database"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/database/pagination"
)

//go:generate mockgen -destination=./mocks/mock_Doa.go -package=mocks github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/business/todos/daos Dao
type Dao interface {
	MigrateDB() error
	GetAllPaginated(
		page *pagination.PageInput,
		sort *models.SortOrder,
		filter *models.Filter,
		projection *models.Projection,
	) ([]*models.Todo, *pagination.PageOutput, error)
	CreateOrUpdate(tt *models.Todo) (*models.Todo, error)
	FindByID(id string, projection *models.Projection) (*models.Todo, error)
}

func NewDao(db database.DB) Dao {
	return &dao{db: db}
}
