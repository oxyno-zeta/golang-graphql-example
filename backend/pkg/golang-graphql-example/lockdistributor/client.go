package lockdistributor

import (
	"context"
	"time"

	"emperror.dev/errors"

	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/config"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/database"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/log"
)

const acquireTimeoutDuration = 30 * time.Second

// ErrLockNotAcquired is returned when a lock cannot be acquired.
var ErrLockNotAcquired = errors.New("lock not acquired")

//go:generate mockgen -destination=./mocks/mock_Service.go -package=mocks github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/lockdistributor Service
type Service interface {
	// Get a lock object (semaphore on string) that can be acquired and release
	GetLock(name string) Lock
	// InitializeAndReload service
	InitializeAndReload(logger log.Logger) error
}

//go:generate mockgen -destination=./mocks/mock_Lock.go -package=mocks github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/lockdistributor Lock
type Lock interface {
	// Acquire lock
	// Deprecated: Use AcquireWithContext instead.
	Acquire() error
	// Acquire lock with context
	AcquireWithContext(ctx context.Context) error
	// Release lock
	Release() error
	// Check if a lock with this name is already taken
	IsAlreadyTaken() (bool, error)
	// Check if the lock is released or lost because of missing heartbeat
	IsReleased() (bool, error)
}

func NewService(cfgManager config.Manager, db database.DB) Service {
	return &service{
		cfgManager: cfgManager,
		db:         db,
	}
}
