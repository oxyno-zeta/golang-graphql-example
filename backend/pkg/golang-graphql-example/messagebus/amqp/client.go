package amqpbusmessage

import (
	"context"
	"time"

	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/common/errors"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/config"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/log"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/metrics"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/signalhandler"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/tracing"
	"github.com/rabbitmq/amqp091-go"
)

var defaultReconnectWaitingDuration = 200 * time.Millisecond
var defaultPublishTimeout = 10 * time.Second
var defaultRetryDelay = time.Second
var tracingPublishOperation = "amqp:publish"
var tracingConsumeOperation = "amqp:consume"

// ErrPublishTimeoutReached is the error thrown when the publish timeout is over.
var ErrPublishTimeoutReached = errors.NewInternalServerError("timeout reached")

// ErrMessageNotJSON is the error thrown when the message utils parse function is called on a non "application/json" message.
var ErrMessageNotJSON = errors.NewInvalidInputError("input haven't the json content type")

// PublishConfigInput represents the publish configuration input.
type PublishConfigInput struct {
	// Exchange is the exchange name where the message is published.
	Exchange string
	// RoutingKey is the published message routing key.
	RoutingKey string
	// Mandatory (see mandatory in AMQP)
	Mandatory bool
	// Immediate (see immediate in AMQP)
	Immediate bool
	// Timeout represents the timeout to wait for publishing a message and have an ack from server including retries.
	// If max timeout is reach, the ErrPublishTimeoutReached is raised.
	// If not set, a default timeout is set to 10 seconds.
	Timeout time.Duration
	// RetryDelay represents the wait delay to consider that a message haven't been sent to server.
	// This can be considered as the timeout between a message is sent and no ack have been detected from server.
	// This will trigger a message publish retry.
	// If not set, a default delay is set to 1 second.
	// Don't go below this limit as a message can takes time to be ack.
	RetryDelay time.Duration
}

// ConsumeConfigInput represents the consume configuration input.
type ConsumeConfigInput struct {
	// QueueName is the queue name for consume.
	QueueName string
	// ConsumerPrefix is the prefix used for the consumer tag in AMQP consumer.
	// The suffix is the hostname.
	ConsumerPrefix string
	// AutoAck
	AutoAck bool
	// Exclusive
	Exclusive bool
	// NoLocal
	NoLocal bool
	// NoWait
	NoWait bool
	// Args
	Args amqp091.Table
	// RetryDelay is the delay between two consume try.
	// This take actions when the channel is disconnected for example.
	RetryDelay time.Duration
	// RequeueOnNackFn is a function that is called to have the requeue flag on a
	// nack response when the message consume is in error.
	// The default value is true is no function is set.
	RequeueOnNackFn func(d *amqp091.Delivery, err error) bool
}

// Client represents the AMQP client.
//
//go:generate mockgen -destination=./mocks/mock_Client.go -package=mocks github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/messagebus/amqp Client
type Client interface {
	// Connect will connect and create channels.
	Connect() error
	// Close will close all channels and connections.
	Close() error
	// Reconnect will handle the close and connect sequence.
	Reconnect() error
	// CancelAllConsumers will cancel all consumers.
	// This must be used in stop management.
	CancelAllConsumers() error
	// Publish will allow to publish a message.
	Publish(
		ctx context.Context,
		messageCfg *amqp091.Publishing,
		publishCfg *PublishConfigInput,
	) error
	// Consume will allow to consumer messages.
	// GetConsumeConfig is a function to allow the support of hot reloading the configuration.
	// Cb is a function that is called each time a message is handled.
	Consume(
		ctx context.Context,
		getConsumeCfg func() *ConsumeConfigInput,
		cb func(ctx context.Context, delivery *amqp091.Delivery) error,
	) error
	// Ping will check connections statuses.
	Ping() error
}

func New(
	logger log.Logger,
	cfgManager config.Manager,
	tracingSvc tracing.Service,
	signalHandlerSvc signalhandler.Client,
	metricsSvc metrics.Client,
) Client {
	return &amqpService{
		logger:           logger,
		cfgManager:       cfgManager,
		tracingSvc:       tracingSvc,
		signalHandlerSvc: signalHandlerSvc,
		metricsSvc:       metricsSvc,
		consumerTags:     []string{},
	}
}
