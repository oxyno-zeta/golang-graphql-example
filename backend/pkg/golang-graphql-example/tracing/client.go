package tracing

import (
	"context"
	"net/http"

	gqlgraphql "github.com/99designs/gqlgen/graphql"
	"github.com/gin-gonic/gin"
	opentracing "github.com/opentracing/opentracing-go"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/config"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/log"
	"gorm.io/gorm"
)

// Service Tracing service.
//
//go:generate mockgen -destination=./mocks/mock_Service.go -package=mocks github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/tracing Service
type Service interface {
	// Reload service.
	Reload() error
	// Get opentracing tracer.
	GetTracer() opentracing.Tracer
	// Http Gin HttpMiddleware to add trace per request.
	HTTPMiddleware(getRequestID func(ctx context.Context) string) gin.HandlerFunc
	// Graphql Middleware.
	GraphqlMiddleware() gqlgraphql.HandlerExtension
	// Get database middleware.
	DatabaseMiddleware() gorm.Plugin
	// StartSpan will return a new trace created from scratch.
	StartTrace(operationName string) Trace
	// StartChildTraceOrTraceFromContext will return a child trace if a trace is found inside
	// the context or a new trace with the operation name.
	StartChildTraceOrTraceFromContext(ctx context.Context, operationName string) Trace
	// ExtractFromTextMapAndStartSpan will extract trace from textmap and start a new one from this one.
	ExtractFromTextMapAndStartSpan(txtMap map[string]string, operationName string) (Trace, error)
	// ExtractFromTextMapAndStartSpan will extract trace from http headers and start a new one from this one.
	ExtractFromHTTPHeaderAndStartSpan(headers http.Header, operationName string) (Trace, error)
}

// Trace structure.
//
//go:generate mockgen -destination=./mocks/mock_Trace.go -package=mocks github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/tracing Trace
type Trace interface {
	// Add tag to trace.
	SetTag(key string, value interface{})
	// Add tags to trace.
	SetTags(tags map[string]interface{})
	// MarkAsError will mark trace as in error.
	MarkAsError()
	// Get a child trace.
	GetChildTrace(operationName string) Trace
	// End the trace.
	Finish()
	// Get the trace ID.
	GetTraceID() string
	// InjectInHTTPHeader will inject span in http header for forwarding.
	InjectInHTTPHeader(header http.Header) error
	// InjectInTextMap will inject span in text map for forwarding.
	InjectInTextMap(textMap map[string]string) error
}

func New(cfgManager config.Manager, logger log.Logger) (Service, error) {
	return newService(cfgManager, logger)
}
