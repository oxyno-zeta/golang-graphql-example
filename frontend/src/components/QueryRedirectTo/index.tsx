import QueryRedirectTo, { Props } from './QueryRedirectTo';

export default QueryRedirectTo;
export type { Props };
