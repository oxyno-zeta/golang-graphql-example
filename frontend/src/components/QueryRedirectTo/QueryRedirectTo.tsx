import React from 'react';
import { useTranslation } from 'react-i18next';
import { useQuery, DocumentNode, QueryHookOptions, OperationVariables } from '@apollo/client';
import { useParams, Params, Navigate, NavigateProps } from 'react-router-dom';
import CenterLoading, { Props as CenterLoadingProps } from '~components/CenterLoading';
import GraphqlErrors, { Props as GraphqlErrorsProps } from '~components/GraphqlErrors';
import NoData, { Props as NoDataTypographyProps } from '~components/NoData';

export interface Props<T, P extends OperationVariables> {
  // Query document to execute.
  query: DocumentNode;
  // Build navigate to function will return a path.
  // If path result is empty or undefined, then "no data" is displayed.
  // If you don't want any "no data" displayed, manage the redirect path
  buildNavigateTo: (params: T | undefined) => string | undefined | null;
  // Build query variables function will return a query variable object.
  buildQueryVariables: (params: Params<string>) => P;
  // Disable center loading subtitle.
  disableCenterLoadingSubtitle?: boolean;
  // Query hook options.
  queryHookOptions?: Omit<QueryHookOptions<T, P>, 'variables'>;
  // No data Typography props.
  noDataTypographyProps?: NoDataTypographyProps;
  // Center loading props.
  centerLoadingProps?: CenterLoadingProps;
  // Graphql Errors props.
  graphqlErrorsProps?: Omit<GraphqlErrorsProps, 'error|errors'>;
  // Navigate props.
  navigateProps?: Omit<NavigateProps, 'to'>;
}

const defaultProps = {
  queryHookOptions: {},
  noDataTypographyProps: {},
  centerLoadingProps: {},
  graphqlErrorsProps: {},
  navigateProps: {},
  disableCenterLoadingSubtitle: false,
};

function QueryRedirectTo<T, P extends OperationVariables>({
  query,
  buildQueryVariables,
  buildNavigateTo,
  disableCenterLoadingSubtitle,
  queryHookOptions,
  noDataTypographyProps,
  centerLoadingProps,
  graphqlErrorsProps,
  navigateProps,
}: Props<T, P>) {
  // Setup translate
  const { t } = useTranslation();
  // Get params
  const params = useParams();

  // Build query variables
  const queryVariables: P = buildQueryVariables(params);

  // Query
  const { data, loading, error } = useQuery<T, P>(query, {
    variables: queryVariables,
    fetchPolicy: 'network-only',
    ...queryHookOptions,
  });

  // Check loading
  if (loading) {
    return (
      <CenterLoading
        subtitle={!disableCenterLoadingSubtitle ? t('common.loadingText') : undefined}
        containerBoxSx={{ margin: '15px 0' }}
        {...centerLoadingProps}
      />
    );
  }

  // Check error
  if (error) {
    return <GraphqlErrors error={error} {...graphqlErrorsProps} />;
  }

  // Build navigate to
  const to = buildNavigateTo(data);

  // Check if to isn't present
  if (!to) {
    return <NoData {...noDataTypographyProps} />;
  }

  return <Navigate to={to} {...navigateProps} />;
}

QueryRedirectTo.defaultProps = defaultProps;

export default QueryRedirectTo;
