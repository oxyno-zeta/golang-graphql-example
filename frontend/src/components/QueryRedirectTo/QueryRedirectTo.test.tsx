import React, { type ReactNode } from 'react';
import { MemoryRouter, Route, Routes } from 'react-router';
import { render, waitFor } from '@testing-library/react';
import { ApolloError, gql } from '@apollo/client';
import { MockedProvider, type MockedResponse } from '@apollo/client/testing';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

import { GraphQLError } from 'graphql';
import QueryRedirectTo from './QueryRedirectTo';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

const QUERY = gql`
  query Q($name: String) {
    fake(name: $name) {
      id
    }
  }
`;

interface Props {
  readonly children: ReactNode;
  readonly mockedResponse: MockedResponse;
}

function TestComponent({ children, mockedResponse }: Props) {
  return (
    <MockedProvider addTypename={false} mocks={[mockedResponse]}>
      <MemoryRouter initialEntries={['/route/fake-name']}>
        <Routes>
          <Route element={<div>Fake</div>} path="/fake" />
          <Route element={<>{children}</>} path="/route/:name" />
        </Routes>
      </MemoryRouter>
    </MockedProvider>
  );
}

describe('QueryRedirectTo', () => {
  it('should display a loading when query is pending', async () => {
    const { container, findByRole } = render(
      <TestComponent
        mockedResponse={{
          request: {
            query: QUERY,
            variables: { name: 'fake-name' },
          },
        }}
      >
        <QueryRedirectTo buildNavigateTo={() => '/fake'} buildQueryVariables={({ name }) => ({ name })} query={QUERY} />
      </TestComponent>,
    );

    expect(container).toMatchSnapshot();

    // Find progressbar
    expect(container).toHaveTextContent('common.loadingText');
    expect(await findByRole('progressbar')).not.toBeNull();
  });

  it('should display a loading without subtitle when query is pending', async () => {
    const { container, findByRole } = render(
      <TestComponent
        mockedResponse={{
          request: {
            query: QUERY,
            variables: { name: 'fake-name' },
          },
        }}
      >
        <QueryRedirectTo
          buildNavigateTo={() => '/fake'}
          buildQueryVariables={({ name }) => ({ name })}
          disableCenterLoadingSubtitle
          query={QUERY}
        />
      </TestComponent>,
    );

    expect(container).toMatchSnapshot();

    // Find progressbar
    expect(container).not.toHaveTextContent('common.loadingText');
    expect(await findByRole('progressbar')).not.toBeNull();
  });

  it('should display an error when an error is raised on query', async () => {
    const { container, findByRole } = render(
      <TestComponent
        mockedResponse={{
          request: {
            query: QUERY,
            variables: { name: 'fake-name' },
          },
          error: new ApolloError({
            graphQLErrors: [
              new GraphQLError('forbidden graphql error', {
                extensions: { code: 'FORBIDDEN' },
              }),
            ],
          }),
        }}
      >
        <QueryRedirectTo buildNavigateTo={() => '/fake'} buildQueryVariables={({ name }) => ({ name })} query={QUERY} />
      </TestComponent>,
    );

    expect(container).toMatchSnapshot();

    // Find progressbar
    expect(await findByRole('progressbar')).not.toBeNull();

    // Now find errors
    expect(container).toHaveTextContent('common.errors');
    expect(container).toMatchSnapshot();
  });

  it('should display no data when no data is returned from query', async () => {
    const { container, findByRole } = render(
      <TestComponent
        mockedResponse={{
          request: {
            query: QUERY,
            variables: { name: 'fake-name' },
          },
          result: {
            data: null,
          },
        }}
      >
        <QueryRedirectTo buildNavigateTo={() => null} buildQueryVariables={({ name }) => ({ name })} query={QUERY} />
      </TestComponent>,
    );

    expect(container).toMatchSnapshot();

    // Find progressbar
    expect(await findByRole('progressbar')).not.toBeNull();

    // Now find errors
    expect(container).toHaveTextContent('common.noData');
    expect(container).toMatchSnapshot();
  });

  it('should redirect to new path on success', async () => {
    const { container, findByRole } = render(
      <TestComponent
        mockedResponse={{
          request: {
            query: QUERY,
            variables: { name: 'fake-name' },
          },
          result: {
            data: { fake: { id: '1' } },
          },
        }}
      >
        <QueryRedirectTo buildNavigateTo={() => '/fake'} buildQueryVariables={({ name }) => ({ name })} query={QUERY} />
      </TestComponent>,
    );

    expect(container).toMatchSnapshot();

    // Find progressbar
    expect(await findByRole('progressbar')).not.toBeNull();

    // Workaround to avoid "react component change without any act called"...
    await waitFor(() => 0);

    // Now find errors
    expect(container).toHaveTextContent('Fake');
    expect(container).toMatchSnapshot();
  });
});
