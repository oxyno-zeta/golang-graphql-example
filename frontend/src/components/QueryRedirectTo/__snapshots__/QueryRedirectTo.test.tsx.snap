// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryRedirectTo should display a loading when query is pending 1`] = `
<div>
  <div
    class="MuiBox-root css-1rt4buw"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-mmopz6-MuiTypography-root"
    >
      common.loadingText
    </h6>
  </div>
</div>
`;

exports[`QueryRedirectTo should display a loading without subtitle when query is pending 1`] = `
<div>
  <div
    class="MuiBox-root css-1rt4buw"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`QueryRedirectTo should display an error when an error is raised on query 1`] = `
<div>
  <div
    class="MuiBox-root css-1rt4buw"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-mmopz6-MuiTypography-root"
    >
      common.loadingText
    </h6>
  </div>
</div>
`;

exports[`QueryRedirectTo should display an error when an error is raised on query 2`] = `
<div>
  <div
    class="MuiBox-root css-z0zjqa"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      class="MuiBox-root css-0"
    >
      <li
        class="MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`QueryRedirectTo should display no data when no data is returned from query 1`] = `
<div>
  <div
    class="MuiBox-root css-1rt4buw"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-mmopz6-MuiTypography-root"
    >
      common.loadingText
    </h6>
  </div>
</div>
`;

exports[`QueryRedirectTo should display no data when no data is returned from query 2`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1xmhd5p-MuiTypography-root"
  >
    common.noData
  </p>
</div>
`;

exports[`QueryRedirectTo should redirect to new path on success 1`] = `
<div>
  <div
    class="MuiBox-root css-1rt4buw"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-mmopz6-MuiTypography-root"
    >
      common.loadingText
    </h6>
  </div>
</div>
`;

exports[`QueryRedirectTo should redirect to new path on success 2`] = `
<div>
  <div>
    Fake
  </div>
</div>
`;
